api/config_entry_discoverychain.go

```diff
@@ -63,6 +65,7 @@ type ServiceRouteHTTPMatchQueryParam struct {
 type ServiceRouteDestination struct {
 	Service               string               `json:",omitempty"`
 	ServiceSubset         string               `json:",omitempty" alias:"service_subset"`
+	Partition             string               `json:",omitempty"`
```

Comment:
This needs to be removed, discovery chain routes can't reference other partitions